File: dailyprogrammer/111easy.c

		#include <stdio.h>
		#include <string.h>
		#include <stdlib.h>

		void unstar(char * str) {
			int len = strlen(str);
			int mask[len + 2];
			char newstr[len+1];
			
			int j;
			for (j=0; j < len + 2; j++)
				mask[j] = 0;

			int i;
			for (i = 1; i < len + 1; i++) {
				if (str[i-1] == '*') {
					mask[i-1] = mask[i] = mask[i+1] = -1;
				}
			}

			int l,s;
			for (l = 0, s = 0; l < len; l++) {
				if (mask[l+1] == 0) {
					newstr[s] = str[l];
					s++;
				}

			}
			newstr[s] = 0;

			printf("%s --> %s\n", str, newstr);
		}

		int main() {

			unstar ("adf*lp");
			unstar ("a*o");
			unstar ("*dech*");
			unstar ("de**po");
			unstar ("sa*n*ti");
			unstar ("abc");

			return 0;
		}

File: dailyprogrammer/111easy.js
/*
 * [11/6/2012] Challenge #111 [Easy] Star delete (self.dailyprogrammer)
submitted 1 month ago* by Cosmologicon Python
Write a function that, given a string, removes from the string any * character, or any character that's one to the left or one to the right of a * character. Examples:
"adf*lp" --> "adp"
"a*o" --> ""
"*dech*" --> "ec"
"de**po" --> "do"
"sa*n*ti" --> "si"
"abc" --> "abc"
Thanks to user larg3-p3nis for suggesting this problem in /r/dailyprogrammer_ideas[1] !

*/

		function unstar(str) {
			return str.replace(/.?\*+.?/g, '');
		}


		console.log(unstar ("adf*lp"))
		console.log(unstar ("a*o"))
		console.log(unstar ("*dech*"))
		console.log(unstar ("de**po"))
		console.log(unstar ("sa*n*ti"))
		console.log(unstar ("abc"))
